### DISCUSSION TITLE
Does anyone know how to endconversation in composer?

### DISCUSSION THREAD

**Discussion by tushaar9027** [microsoft/BotFramework-Composer#8923] [Q&A]
Does anyone know how to endconversation in composer?

**Comment by jpcrodrigues** [comment]
Why don't you use end dialog action?

![image](https://user-images.githubusercontent.com/23581978/152379294-4f34348f-e842-4d87-bdac-1b9d6aaf75f1.png)

**Comment by tushaar9027** [comment]
Hi @RJayPC , I need to pass end of conversation event to directline , i tried with end dialog but in directline recieve activity endpoint it comes as normal activity nowwhere it is speciefied its the end of conversation.

**Comment by jpcrodrigues** [comment]
I've been looking arround in the [documentation](https://docs.microsoft.com/en-us/composer/concept-events-and-triggers?tabs=v2x) and seems that you must create a new trigger with that specific kind of activitiy.

To do that, select a dialog and add new trigger. A popup will be shown, select the options as ilustrated in the image bellow:
![image](https://user-images.githubusercontent.com/23581978/152386186-a67ac4a0-59a1-4905-8a13-652a127df3ef.png)

**Reply by tushaar9027** [reply]
Yes , i am using this only to catch the endofconversation event raised from directline send activities endpoint....

In my case i am connecting to a third party using my custom api that api in turn is calling my composer bot using directline api.

Basically a third party primary bot is consuming my bot as a service by the custom endpoint created by me.
The issue arises  when i need to pass endodconversation flag to them but i don't get anything from composer on enddialog block in activities array of directline.
